### DISCUSSION TITLE
Allow limit to the size of forecast results

### DISCUSSION THREAD

**Discussion by gabe-sky** [weather-gov/api#216] [Q&A]
**Is your feature request related to a problem? Please describe.**
I'm trying to make requests of the forecast/hourly endpoint, from a RAM-limited IoT device.  Currently, the reply is so big that my device runs out of space to hold it.

**Describe the solution you'd like**
I'd love to have an optional parameter to limit the reply to just X hours.

**Describe alternatives you've considered**
I could parse it line by line, but any change to the order of lines in the response would break my workaround.

**Comment by RenateUSB** [comment]
How about disabling indentation of JSON?
That would get rid of about half the characters.

While you're at it, how about limiting the number of decimal on values?
Having 16 decimal places on temperature seems excessive?
    "value": 7.7777777777777777

How about merging periods?
    {
        "validTime": "2021-04-30T00:00:00+00:00/PT6H",
        "value": 0
    },
    {
        "validTime": "2021-04-30T06:00:00+00:00/PT6H",
        "value": 0
    }
How about omitting values like snowfallAmount when it's zero?
